hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/slowlog/LogEventHandler.java

```diff
@@ -53,12 +55,28 @@
 class LogEventHandler implements EventHandler<RingBufferEnvelope> {
 
   private static final Logger LOG = LoggerFactory.getLogger(LogEventHandler.class);
+  private static final int SYS_TABLE_QUEUE_SIZE = 1000;
```

Comment:
My ask was do u want to put a new config similar to the ring buffer's rather than hard coded 1000.
After discussion, it looks like the confusion mostly coming because of these Q sizes and its value is not really going with its intent.  Ideally the ringbuffer Q is supposed to hold the slow logs for much longer time. The table Q is cleared/reduced in every 10 mins. So ideally the size of the table Q <= ring buffer Q size.  Then all confusions will go.
I would suggest we do this way.
We can have a new config like 'hbase.regionserver.slowlog.ringbuffer.size' for the table Q size. Its default value can be the value of hbase.regionserver.slowlog.ringbuffer.size' itself.   This 2 Q system will allow user to even have a tableQ size > ringbuffer Q size by tuning the config.